### PR TITLE
auto mode for chat timerange picker

### PR DESCRIPTION
**Pull request by philipperolet**
Draft PR for sharing / experimentation
Screenshots below

Current issues whose path to improvement are unclear:
- Falls back on "all time" often. Fine, but the user sees it in the query loading so a bit dissapointing
- might encourage users to ask "what did I say about XX 3 weeks ago?" --- which is not handled at present and cannot be handled simply on a UX perspective
- Examples of okayish behaviour
  - "company roadmap Q1 2023?" => "all"
- Examples of wrong behaviour
  - "In january, was there a meeting in which we discussed the hiring targets?" => "month" (question asked with date 31 may)
  - "Describe the incident in production that happened 3 weeks ago" => "3 weeks" (invalid value)
  - "What was the revenue target for last year?" => "year" ( asked with date 31/05; the targets were probably set Q1 last year or even Q4 of the preceding year so they will likely be missed)

### CODE REVIEW

**Comment by philipperolet** [comment]
image

**Comment by philipperolet** [comment]
image

File: front/types/chat.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { TimeRangeId } from "@app/components/use/ChatTimeRangePicker";
```
Comment (spolu) [inline front/types/chat.ts:1]: Nope :)

Def not want to have types imported from a component in here. The idea of these types is to be our source of truth for types that go through the network interface.

Def want to move the TimeRangeId here instaed of the other way around

**Review by spolu — CHANGES_REQUESTED** [review]
Looks great! Left one comment that needs addressing.

Otherwise, I suggest we move to a more flexible time picker when generated (fine if the user can't set 3 weeks but if the model can I think as a first step)

See: https://github.com/dust-tt/dust/blob/main/front/components/use/GensTimeRangePicker.tsx

Won't be the same as we need the `auto` and stuff but I think the timeRange model + associated functions call here: https://dust.tt/w/78bda07b39/a/699bfebc61 might be more expressive.

Can be tackled in a subsequent PR if you prefer :+1:

**Comment by philipperolet** [comment]
> Looks great! Left one comment that needs addressing.
> 
> Otherwise, I suggest we move to a more flexible time picker when generated (fine if the user can't set 3 weeks but if the model can I think as a first step)
> 
> See: https://github.com/dust-tt/dust/blob/main/front/components/use/GensTimeRangePicker.tsx
> 
> Won't be the same as we need the `auto` and stuff but I think the timeRange model + associated functions call here: https://dust.tt/w/78bda07b39/a/699bfebc61 might be more expressive.
> 
> Can be tackled in a subsequent PR if you prefer +1

Yes agree flexibility can be tried and may solve a few issues mentioned above; that being said, would recommend not pushing to production until we have an eval

**Comment by spolu** [comment]
Right now the model just choose 1 week 1 month 1 year right?

**Comment by philipperolet** [comment]
Yes, or all time

**Comment by spolu** [comment]
Roger that. Awaiting update for another pass of review :+1:

**Pull request closed without merging**